### PR TITLE
NEPT-2862 -- Impose EU Login 2 FA for non whitelisted users roles.

### PR DESCRIPTION
**Pull request by mehdi0712**
### Description

Impose EU Login 2 FA for non whitelisted users.

### Change log

- Added: settings.php variable storing whitelisted roles, EcasDenieAuth class that centralize login authentication checks process, new warning page informing user about his account requirements in case of login denial.

- Changed: ecas_login_page() when a user is correctly logged in do not redirect him to a warning page, _ecas_warning_goto() logic moved, _ecas_session_end() logic moved

### Checklist

- [ ] Check if there are hook_updates and they are documented
- [ ] Add right labels to indicate in the devops ticket the commands to run
- [ ] Remove symlinks if it's a module removal

### CODE REVIEW

File: profiles/common/modules/custom/ecas/includes/ecas.inc [hunk 1]
```diff
@@ -1350,6 +1323,13 @@ EOF;
 EOF;
       break;
 
+    case ECAS_WARNING_MESSAGE_TFA_REQUIRED:
+      $value = <<< EOF
+      You just logged in using a one factor authentication method (I.e. Password) when your account requires a stronger authentication.
+      Please reiterate your authentication by choosing a more secured authentication method (I.e. Mobile Phone + SMS).
```
Comment (tomv-be) [inline profiles/common/modules/custom/ecas/includes/ecas.inc]:
```suggestion
      Please renew your authentication by choosing a more secure authentication method (e.g. Mobile Phone + SMS).
```
I wasn't sure about "reiterate", but "renew" is probably easier to understand regardless.

File: profiles/common/modules/custom/ecas/includes/ecas.inc [hunk 2]
```diff
@@ -1350,6 +1323,13 @@ EOF;
 EOF;
       break;
 
+    case ECAS_WARNING_MESSAGE_TFA_REQUIRED:
+      $value = <<< EOF
+      You just logged in using a one factor authentication method (I.e. Password) when your account requires a stronger authentication.
```
Comment (tomv-be) [inline profiles/common/modules/custom/ecas/includes/ecas.inc]:
```suggestion
      You just logged in using a one-factor authentication method (i.e. Password), when your account requires a stronger authentication.
```

File: profiles/common/modules/custom/ecas/includes/EcasDenieAuth.inc [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+/**
+ * @file
+ * Contains EcasDenieAuth.
+ */
+
+/**
+ * Contains logic to check and manage authentication denie.
+ */
+class EcasDenieAuth {
```
Comment (tomv-be) [inline profiles/common/modules/custom/ecas/includes/EcasDenieAuth.inc:11]: In the class name, it should probably be either "deny" or "denied"? "Denie" is not a word, I think? But we'd need to make sure it's changed everywhere consistently, though. We don't wanna cause problems my trying to fix this. If too much trouble, leave it.

**Review by tomv-be — CHANGES_REQUESTED** [review]
Functions very well. I left some nitpicky suggestions with regard to spelling.

**Comment by Fefaine** [comment]
Replaced by 
Management wants to restrict to 2.6
https://github.com/ec-europa/platform-dev/pull/2948

**Pull request closed without merging**